crates/aptos-dkg/src/pvss/chunked_elgamal_field/chunked_elgamal.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::algebra::homomorphism::{self, FixedBaseMSM};
+use ark_ec::{pairing::Pairing, CurveGroup, VariableBaseMSM};
+
+#[allow(non_snake_case)]
+pub struct Map<'a, E: Pairing> {
+    pub g_1: &'a E::G1Affine,
+    pub h_1: &'a E::G1Affine,
+    pub ek: &'a [E::G1Affine],
+}
+
+type MatrixVectorPair<T> = (Vec<Vec<T>>, Vec<T>);
+type MSMInputVec<'a, E> = MatrixVectorPair<(
+    Vec<<Map<'a, E> as FixedBaseMSM>::Base>,
```

Comment:
I'd add typedefs for `Map<'a, E> as FixedBaseMSM>::Base` and `Map<'a, E> as FixedBaseMSM>::Scalar` to make this a bit more readable.